lib/ExWebhook/web/principal_data_plug.ex

```diff
@@ -0,0 +1,35 @@
+defmodule PrincipalDataPlug do
+  @moduledoc """
+  A plug for extracting and assigning principal data from a request's
+  authorization token.
+
+  It expects a JWT in the "Bearer" token format. If a valid token is
+  found, it assigns a `PrincipalData` struct to the connection.
+  Otherwise, it assigns a default `PrincipalData` struct.
+  """
+  import Plug.Conn
```

Comment:
I don't think we need to track webhook delete events authors, beside, this is too much specific, as this project does not even handle auth and should be used by third parties.